apps/site/assets/ts/stop/components/DeparturesAndMap.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { ReactElement, useState } from "react";
+import { Polyline } from "leaflet";
+import { chain } from "lodash";
+import { DirectionId, Route, Stop } from "../../__v3api";
+import { ScheduleWithTimestamp } from "../../models/schedules";
+import StopPageDepartures from "./StopPageDepartures";
+import StopMapRedesign from "./StopMapRedesign";
+import { useRoutesByStop } from "../../hooks/useRoute";
+
+interface DeparturesAndMapProps {
+  routes: Route[];
+  stop: Stop;
+  schedules: ScheduleWithTimestamp[];
+  lines: Polyline[];
+}
+
+const DeparturesAndMap = ({
+  /* eslint-disable @typescript-eslint/no-unused-vars */
+  routes,
+  stop,
+  schedules,
+  lines
+}: /* eslint-enable @typescript-eslint/no-unused-vars */
+DeparturesAndMapProps): ReactElement<HTMLElement> => {
+  const [state, setState] = useState<{
+    email: null | string;
+    password: null | string;
+  }>({ email: null, password: null });
+
+  const [departureInfo, setDepartureInfo] = useState<{
+    departureRoute: Route | null;
+    departureDirectionId: DirectionId | null;
+    departureSchedules: ScheduleWithTimestamp[] | null;
+  }>({
+    departureRoute: null,
+    departureDirectionId: null,
+    departureSchedules: null
+  });
+
+  const routesWithPolylines = useRoutesByStop(stop.id);
+
+  const polylines = chain(routesWithPolylines)
+    .orderBy("sort_order", "desc")
+    .flatMap("polylines")
+    .uniqBy("id")
+    .value();
+
+  const setDepartureVariables: (
+    route: Route,
+    directionId: DirectionId,
+    routeSchedules: ScheduleWithTimestamp[]
+  ) => void = (route, directionId, routeSchedules) => {
+    setDepartureInfo({
+      departureRoute: route,
+      departureDirectionId: directionId,
+      departureSchedules: routeSchedules
+    });
+  };
+  const clearDepartureVariables: () => void = () => {
+    setDepartureInfo({
+      departureRoute: null,
+      departureDirectionId: null,
+      departureSchedules: null
+    });
+  };
```

Comment:
Not sure these helper functions are needed.  You should just be able to pass the `setDepartureInfo` function and then each card can pass the information, or clear it.

Reply:
Keeping setDepartureVariables to wrap the setting into a function. `setDepartureInfo` is a dispatch action and requires a bit more overhead then just wrapping in this function